### PR TITLE
Require type indices on record declarations

### PR DESCRIPTION
**Pull request by CodaFi**
The empty record declaration is inhabited by the empty record
constructor

```silt
record True : Type where

tt : True
tt = record {}
```

### CODE REVIEW

File: Sources/Crust/Parser.swift [hunk 1]
```diff
@@ -407,7 +407,21 @@ extension Parser {
     let recordTok = try consume(.recordKeyword)
     let recName = try parseIdentifierToken()
     let paramList = try parseTypedParameterList()
-    let indices = peek() == .colon ? try parseTypeIndices() : nil
+    let indices: TypeIndicesSyntax
+
+    // If we see a semicolon or 'where' after the identifier, the
+    // user likely forgot to provide indices for this data type.
+    // Recover by inserting `: Type`.
+    if [.semicolon, .whereKeyword].contains(peek()) {
```
Comment (harlanhaskins) [inline Sources/Crust/Parser.swift]: Please factor this into a helper function

**Review by harlanhaskins — CHANGES_REQUESTED** [review]

File: Sources/Crust/Parser.swift (hunk 1 above)
Comment (CodaFi) [reply Sources/Crust/Parser.swift]: Okay if I just fold it into `parseTypeIndices` itself?

Comment (harlanhaskins) [reply Sources/Crust/Parser.swift]: Sounds good.

Comment (CodaFi) [reply Sources/Crust/Parser.swift]: Done

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**